enhancements/monitoring/restart-metrics.md

```diff
@@ -0,0 +1,127 @@
+---
+title: restart-metrics
+authors:
+  - "@rphillips"
+reviewers:
+  - "@smarterclayton"
+  - "@sjenning"
+  - "@mrunalp"
+approvers:
+  - "@brancz"
+  - "@bparees"
+  - "@squat"
+  - "@s-urbaniak"
+  - "@metalmatze"
+  - "@paulfantom"
+  - "@LiliC"
+  - "@pgier"
+  - "@simonpasquier"
+creation-date: 2020-03-23
+last-updated: 2020-03-23
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Service Restart Metrics
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+There is not any insight with systemd units restarting. Metrics of this sort are
+important to know if Kubelet, crio, or other system services are crashing and
+restarting. Restart metrics are going to be extremely important since the
+Kubelet is going to be changed to exit on a crash, thus restarting. Previous
+behavior of the Kubelet is to recover() from the crash and not exit.
+
+## Motivation
+
+Systemd unit restart metrics of a unit are vitally important for system
+administrators to see the health of the Kubelet, crio, and other vitally important
+system services.
+
+### Goals
+
+- Systemd unit restart metrics need to be propogated through to monitoring, and
+  the alerting system to alert system administrators to problems within the
+  system.
+
+### Non-Goals
+
+- Defining alerts is not in the scope of this proposal. Alerts can be defined
+  once we see some sample cluster data.
+
+### Current Behavior
+
+node_exporter is running in a non-privileged container - as user `nobody` - and
```

Comment:
per our conversation, you might point out that is it not being a non-root user that is the probably.  it is the selinux labelling given to processes in non-privileged containers.  we are looking to run this in a privileged container as a non-root user.

Reply:
good idea... fixed.